File: coding-test-1/practice-problems/solutions/03-sum-nums.rb
# Write a method that takes in an integer `num` and returns the sum of
# all integers between zero and num, up to and including `num`.
#
# Difficulty: easy.

def sum_nums(num)
  result = 0

  i = 0
  while i <= num
    result += i
    i += 1
  end

  return result
end

puts("sum_nums(1) == 1: #{sum_nums(1) == 1}")
puts("sum_nums(2) == 3: #{sum_nums(2) == 3}")
puts("sum_nums(3) == 6: #{sum_nums(3) == 6}")
puts("sum_nums(4) == 10: #{sum_nums(4) == 10}")
puts("sum_nums(5) == 15: #{sum_nums(5) == 15}")

File: coding-test-1/practice-problems/solutions-python/03-sum-nums.py
# Write a method that takes in an integer `num` and returns the sum of
# all integers between zero and num, up to and including `num`.
#
# Difficulty: easy.
import unittest


def sum_nums(num):
    total = 0
    for i in range(num + 1):
        total += i
    return total

# These are tests to check that your code is working. After writing
# your solution, they should all run.


class SumNums(unittest.TestCase):

    def test_one_sum(self):
        self.assertEqual(sum_nums(1), 1)

    def test_two_sum(self):
        self.assertEqual(sum_nums(2), 3)

    def test_three_sum(self):
        self.assertEqual(sum_nums(3), 6)

    def test_four_sum(self):
        self.assertEqual(sum_nums(4), 10)

    def test_five_sum(self):
        self.assertEqual(sum_nums(5), 15)

if __name__ == '__main__':
    unittest.main()
